src/qililab/instruments/qblox/qblox_draw.py

```diff
@@ -0,0 +1,534 @@
+
+        Args:
+            param (dictionary): parameters of the bus (IF, phase, offset, hardware modulation).
+
+        Returns:
+            Sets the phase as 0 in the param dictionary.
+        """
+        if param["phase_new"]:
+            param["phase"][-1] = 0
+        else:
+            param["phase"].append(0)
+        param["phase_new"] = True
+        return param
+
+    def _handle_add_draw(self, register, program_line):
+        """Updates the register dictionary when a line of the parsed program has a move command.
+
+        Args:
+            register (dictionary): registers of the Q1ASM.
+            program_line (tuple): line of the Q1ASM program parsed with a freq or phase instruction.
+
+        Returns:
+            Updated register.
+        """
+        a, b, destination = program_line[1].split(", ")
+        register[destination] = self._get_value(a, register) + self._get_value(b, register)
+        return register
+
+    def _get_value(self, x, register):
+        """
+        Args:
+            x (string|int|float): desired value or variable.
+            register (dictionary): registers of the Q1ASM.
+
+        Returns:
+            If the input is a number it returns a float; if the input is a key of the registery it returns th evalue as a float
+        """
+        if x is not None:
+            if x.isdigit():
+                return float(x)
+            if x in register:
+                return float(register[x])
+        return None
+
+    def _parse_program(self, sequences):
+        """Parses the program dictionary of the sequence.
+        Args:
+            sequences (dictionary): for each bus, it is made up of nested dictionaries (waveforms,weights,acquisitions and program)
+
+        Returns:
+            Returns seq_parsed_program with each section (setup/main) as a list of tuple (instruction, numerical value, (tuple of loop label), index)
+            Example: Q1ASM has play 0,1,30 -  the equivalent seq_parsed_program is (play,'0,1,30', (),1)
+                where 1 is the index and () is a tuple of the loop labels
+                ie: (avg_0, loop_0) meaning the current isntruction is part of 2 loops, avg_0 as top and loop_0 as nested.
+
+        """
+        seq_parsed_program = {}
+        for bus in sequences:  # Iterate through the bus of the sequences
+            sequence = sequences[bus].todict()
+            program_line = sequence["program"].split("\n")
+            processed_lines = []
+
+            # parse all lines of qprogram and deal with special case of reset phase
+            for line in program_line:
+                stripped_line = line.strip()
+                if stripped_line.startswith("reset_ph"):
+                    processed_lines.append("reset_ph             2")
+                else:
+                    processed_lines.append(stripped_line)
+            line_data = "\n".join(processed_lines)
+            pattern = r"(\w+):|(\w+)(?:\s+([^\n]*))?"
```

Comment:
I'm bad with regexp, would be nice a comment explaining what are we parsing here, maybe an example of a typical line